src/sourcekit-lsp/extension.ts

```diff
@@ -0,0 +1,56 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VSCode Swift open source project
+//
+// Copyright (c) 2021 the VSCode Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VSCode Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+'use strict';
+import * as vscode from 'vscode';
+import * as langclient from 'vscode-languageclient/node';
+import { activateInlayHints } from './inlayHints';
+
+export async function activate(context: vscode.ExtensionContext): Promise<void> {
+    const config = vscode.workspace.getConfiguration('sourcekit-lsp');
+
+    const sourcekit: langclient.Executable = {
+        command: config.get<string>('serverPath', 'sourcekit-lsp'),
+        args: config.get<string[]>('serverArguments', [])
+    };
+
+    const toolchain = config.get<string>('toolchainPath', '');
+    if (toolchain) {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        sourcekit.options = { env: { ...process.env, SOURCEKIT_TOOLCHAIN_PATH: toolchain } };
+    }
+
+    const serverOptions: langclient.ServerOptions = sourcekit;
+
+    let clientOptions: langclient.LanguageClientOptions = {
+        documentSelector: [
+            'swift',
+            'cpp',
```

Comment:
We probably only need `swift` here, as we don't activate the extension for the other file types.

Reply:
We should activate the extension for all languages that SwiftPM supports. Building something like JWTKit which has a ton of code in would be hard if we didn't have that enabled

Reply:
The SourceKit-LSP extension also has activation events for these languages, but these haven't been imported yet. Would you also want those?

I'm just worried this may turn out to be annoying for developers also working on C/C++ projects. I did some work in C last year, and I always had to disable SourceKit-LSP because it kept popping up errors in a valid C project.

Reply:
This is partly related to #21 and others but we probably need to try and be smart about activating the extension. The current Swift extension throws up errors every time you open _any_ workspace and it can't detect a Package.swift. If we make it clever enough so that the extension (and SourceKit LSP and others) are only activated on detection of a top level manifest (plus whatever we decide for #21) that should stop it being a problem